morse_decoder.rb

```diff
@@ -0,0 +1,37 @@
+def decode_char(morse_code)
+  # Define a hash of Morse code characters and their corresponding uppercase letters
+  morse_dic = {
+    '.-' => 'A', '-...' => 'B', '-.-.' => 'C', '-..' => 'D', '.' => 'E',
+    '..-.' => 'F', '--.' => 'G', '....' => 'H', '..' => 'I', '.---' => 'J',
+    '-.-' => 'K', '.-..' => 'L', '--' => 'M', '-.' => 'N', '---' => 'O',
+    '.--.' => 'P', '--.-' => 'Q', '.-.' => 'R', '...' => 'S', '-' => 'T',
+    '..-' => 'U', '...-' => 'V', '.--' => 'W', '-..-' => 'X', '-.--' => 'Y',
+    '--..' => 'Z'
+  }
+  # Return the corresponding uppercase letter for the given Morse code character
+  morse_dic[morse_code]
+end
+
+def decode_word(morse_word)
+  str = ''
+  splited_arr = morse_word.split
+  splited_arr.each do |character|
+    str += decode_char(character)
+  end
+  str
+end
+
+def decode_sentence(morse_sent)
+  str = ''
+  splited_arr = morse_sent.split('   ')
+  splited_arr.each do |sentence|
+    str += decode_word(sentence)
+    str += ''
+  end
+  str
+end
```

Comment:
- [ ] Your sentence decode method does not work as expected. The `str += ''` on line 29 should be `str += ' '`, with a space

  ![image](https://github.com/KUBAHO3/decode-morse-message/assets/36542986/052b455f-6810-4c9c-a691-3dda102e5d03)

- [ ] After fixing that we also have a whitespace at the end of the lines which we need to eliminate. We can do that by adding `#strip` method to the returned string on line 31:
  ```
  str.strip
  ```

  ![image](https://github.com/KUBAHO3/decode-morse-message/assets/36542986/01ffd745-7a31-4c7a-b65e-8d9d7cff62c4)